### ISSUE TITLE
Add .PreBuildDependencies to Executable()

### ISSUE THREAD

**Issue by Leandros** [fastbuild/fastbuild#273]
~~The `Copy()` will not be executed if it's put as a `PreBuildDependency` on a `Executable()` node.
Putting it on the corresponding `ObjectList()` (for example) works fine.~~

~~FASTBuild v0.93 (but doesn't work on earlier versions, too)~~

Yeah, I'm an idiot. `Executable()` has no `PreBuildDependencies`. Would be great to be added, though!

**Comment by ffulin** [comment]
I did some refactoring to Executable/DLL. As a follow up, I added PreBuildDependencies to both of those in (https://github.com/fastbuild/fastbuild/commit/6b3443758dacbc18d17032ef1c5a136637e57c91).

This will be in the next release (v0.95). I'll leave this issue open until that version is released.

One thing that should be noted is that .PreBuildDependencies are special compared to normal dependencies. FASTBuild will ensure they are up-to-date before building the target, but if any of the PreBuildDependencies change, it won't cause the target to rebuild. This allows you to have things that always run (like code generation) and not have the code gen itself cause an unnecessary rebuild.
